pkg/controller/common/reconciler.go

```diff
@@ -30,16 +30,32 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 
 	exposed := ci.GetSpec().Visibility == networkingv1alpha1.IngressVisibilityExternalIP
 	if exposed {
+		listOpts := &client.ListOptions{
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				networking.IngressLabelKey: ci.GetName(),
```

Comment:
I'd add the Knative route labels we add to the routes here as well as this is again prone to errors because it's not namespaced properly. For routes we add a proper namespace label though.